.github/workflows/run-tests.yml

```diff
@@ -145,21 +176,46 @@ jobs:
         run: |
           docker compose -f ${{ env.DOCKER_COMPOSE_FILE }} down -v
 
+
+  set-up-functional-test:
+    if: always()
+    name: Set up functional test
+    runs-on: ubuntu-20.04
+    outputs:
+      shard_indices: ${{ steps.generate_output.outputs.shard_indices }}
+      db_names: ${{ steps.generate_output.outputs.db_names }}
+      total_shards: ${{ steps.generate_output.outputs.shards }}
+      single_test_args: ${{ steps.generate_output.outputs.single_test_args }}
+      functional_test_timeout: ${{ steps.generate_output.outputs.timeout }}
+    steps:
+      - id: generate_output
+        run: |
+          shards=3
+          db_names="['cass_es', 'sqlite', 'mysql8', 'postgres12', 'postgres12_pgx']"
```

Comment:
I think you need to invert the quotes. JSON strings are always with double-quote.
Also, why have the `db_names` here instead of in the matrix directly?